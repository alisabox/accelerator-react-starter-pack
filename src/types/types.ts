import { Action } from 'redux';
import { ThunkAction}  from 'redux-thunk';
import { AxiosInstance } from 'axios';
import { RootState } from '../store/reducers/root-reducer';
import { SeachOptions } from '../const/const';

export type GuitarType = {
  id: number;
  name: string;
  vendorCode: string;
  type: string;
  description: string;
  previewImg: string;
  stringCount: number;
  rating: number;
  price: number;
}

export type CommentType = {
  id: string;
  userName: string;
  advantages: string;
  disadvantages: string;
  comment: string;
  rating: number;
  createAt: Date;
  guitarId: number;
}

export type FilterSettingsType = {
  minPrice: number;
  maxPrice: number;
}

export type GuitarAndCommentsType = GuitarType & {
  comments: CommentType[],
}

export type guitarsStateType = {
  guitarsAndComments: GuitarAndCommentsType[],
  guitarsPerPage: GuitarAndCommentsType[],
  searchResult: GuitarType[],
}

export type filterStateType = {
  filterURLOptions: SeachOptionsType,
  minPrice?: number,
  maxPrice?: number,
}

export type SeachOperatorsBiasedType = {
  [key: string]: string;
}

export type SortOperatorsType = {
  [key: string]: string;
}


export type FilterParamsType = {
  filters: string,
  sort: string,
  order: string,
  pageSearch: string,
  page: string,
}

export type SeachOptionsType = {
  [SeachOptions.PriceMin]: string,
  [SeachOptions.PriceMax]: string,
  [SeachOptions.Acoustic]: boolean,
  [SeachOptions.Electric]: boolean,
  [SeachOptions.Ukulele]: boolean,
  [SeachOptions.FourStrings]: boolean,
  [SeachOptions.SixStrings]: boolean,
  [SeachOptions.SevenStrings]: boolean,
  [SeachOptions.TwelveStrings]: boolean,
}

export type State = RootState;
export type ThunkActionResult<R = Promise<void>> = ThunkAction<R, State, AxiosInstance, Action>;
export type ReselectType<S> = (state: State) => S;
