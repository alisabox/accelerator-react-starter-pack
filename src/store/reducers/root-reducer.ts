import { combineReducers } from 'redux';
import { filterReducer } from './filter/filter-reducer';
import { guitarsReducer } from './guitars/guitars-reducer';

export enum NameSpace {
  Guitars = 'Guitars',
  Filter = 'Filter',
}

export const rootReducer = combineReducers({
  [NameSpace.Guitars]: guitarsReducer,
  [NameSpace.Filter]: filterReducer,
});

export type RootState = ReturnType<typeof rootReducer>;
