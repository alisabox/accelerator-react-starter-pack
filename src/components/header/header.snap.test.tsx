import { Action } from 'redux';
import thunk, { ThunkDispatch } from 'redux-thunk';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { configureMockStore } from '@jedmao/redux-mock-store';
import { createMemoryHistory } from 'history';
import Header from './header';
import { guitars } from '../../mocks/mocks';
import { createAPI } from '../../services/api';
import { State } from '../../types/types';
import { NameSpace } from '../../store/reducers/root-reducer';

const mockData = guitars;

const history = createMemoryHistory();
const api = createAPI();
const middlewares = [thunk.withExtraArgument(api)];
const mockStore = configureMockStore<
    State,
    Action,
    ThunkDispatch<State, typeof api, Action>
  >(middlewares);

const store = mockStore({
  [NameSpace.Guitars]: {
    guitarsAndComments: [],
    guitarsPerPage: [],
    searchResult: mockData,
  },
});

describe('Component: Header', () => {
  test('should render correctly vs. snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Router history={history}>
          <Header />
        </Router>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
