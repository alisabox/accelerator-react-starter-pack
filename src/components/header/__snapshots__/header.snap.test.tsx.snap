// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Header should render correctly vs. snapshot 1`] = `
<div>
  <header
    class="header"
    id="header"
  >
    <div
      class="container header__wrapper"
    >
      <a
        class="header__logo logo"
        href="/#"
      >
        <img
          alt="Логотип"
          class="logo__img"
          height="70"
          src="./img/svg/logo.svg"
          width="70"
        />
      </a>
      <nav
        class="main-nav"
      >
        <ul
          class="main-nav__list"
        >
          <li>
            <a
              class="link main-nav__link link--current"
              href="/#"
            >
              Каталог
            </a>
          </li>
          <li>
            <a
              class="link main-nav__link"
              href="/#"
            >
              Где купить?
            </a>
          </li>
          <li>
            <a
              class="link main-nav__link"
              href="/#"
            >
              О компании
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="form-search"
      >
        <form
          class="form-search__form"
        >
          <button
            class="form-search__submit"
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="form-search__icon"
              height="15"
              width="14"
            >
              <use
                xlink:href="#icon-search"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              Начать поиск
            </span>
          </button>
          <input
            autocomplete="off"
            class="form-search__input"
            id="search"
            placeholder="что вы ищите?"
            type="text"
            value=""
          />
          <label
            class="visually-hidden"
            for="search"
          >
            Поиск
          </label>
        </form>
        <ul
          class="form-search__select-list "
        >
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Честер Bass
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            CURT Z300
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Roman LX
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            CURT T300
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Dania Super
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Честер WX
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Dania VX
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Честер Plus
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Виолана 300
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            CURT Clasic
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            CURT Z250
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Честер 7X
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Честер 6V
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Виолана Mix
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Виолана 250x
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Фабио Лайт
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Фабио L100
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Liana Z200
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Liana Z100
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Liana Z300
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Roman RX
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Roman TX
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Dania U100
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Dania WR
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Dania LE
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Mirana V10
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            Виолана Mini
          </li>
        </ul>
      </div>
      <a
        aria-label="Корзина"
        class="header__cart-link"
        href="/#"
      >
        <svg
          aria-hidden="true"
          class="header__cart-icon"
          height="14"
          width="14"
        >
          <use
            xlink:href="#icon-basket"
          />
        </svg>
        <span
          class="visually-hidden"
        >
          Перейти в корзину
        </span>
        <span
          class="header__cart-count"
        >
          2
        </span>
      </a>
    </div>
  </header>
</div>
`;
